### PR TITLE
R2s script add test

### PR DESCRIPTION
**Pull request by zxkjack123**
This PR provides test functions and files for r2s script using a given simple geometry for both voxel and sub-voxel r2s. It is a solution for issue #34 .

Major changes:
- Add test functions in tests/test_r2s.py
- Add a soft link from pyne/r2s.py to tests/r2s.py
- Add a series of files used by the test functions

### CODE REVIEW

**Comment by zxkjack123** [comment]
This PR fails the 'build_py2' test because the error of import dagmc. The reason of the error is that 'build_py2' is a build without the options of '--moab' and '--dagmc'.
@gonuke @pshriwise I think we can ignore the test failure of 'build_py2' for this PR, as it requires pymoab, how can we expect that it works without pymoab?

File: tests/test_r2s.py [hunk 1]
```diff
@@ -388,3 +390,68 @@ def test_irradiation_setup_unstructured_nondef_tag():
...
+    
+    assert_equal(f1, True)
+    assert_equal(f2, True)
+    assert_equal(f3, True)
+    assert_equal(f4, True)
+    assert_equal(f5, True)
+
+
+def test_r2s_script():
+    # test voxel r2s
```
Comment (gonuke) [inline tests/test_r2s.py:488]: add a `try` block to import dagmc and if it fails, skip the test  - see `test_partisn.py` for an example

**Review by gonuke — CHANGES_REQUESTED** [review]
There should be separate tests for `step1` and `step2`

`step2` test should depend on expected results of `step1` test rather than observed results of `step1` test

**Comment by gonuke** [comment]
Conceptually:
* move testing of `step1` results before running `step2`
* put all `step2` actions in a separate function
* begin `step2` function by copying expected results of `step1` to overwrite observed results

**Comment by gonuke** [comment]
I propose we defer this and PR it into pyne/pyne once we have merged pymoab

**Comment by zxkjack123** [comment]
An error founded when running unstructured r2s step1. Error message:
```
Traceback (most recent call last):
  File "r2s.py", line 227, in <module>
    main()
  File "r2s.py", line 222, in main
    step1()
  File "r2s.py", line 149, in step1
    tag.delete()
  File "/root/.local/lib/python2.7/site-packages/pyne/mesh.py", line 111, in delete
    del self[:]
  File "/root/.local/lib/python2.7/site-packages/pyne/mesh.py", line 188, in __delitem__
    raise AttributeError(msg)
AttributeError: the material property tag 'atoms_per_molecule' may not be deleted
```
It seems that the `tag.delete()` is trying to delete a tag named `atoms_per_molecule`, then the error raised. I made this commit to enable others to repeat this error and debug with example files.

The reason of this error is that in voxel/sub-voxel R2S, the tag contains only NativeMeshTag. However, there are other types of tags such as MaterialPropertyTag.

This bug is fixed in #68 .

**Comment by zxkjack123** [comment]
Rebase after the merge of #68.

**Comment by zxkjack123** [comment]
Put this PR to pyne/pyne after pymoab get merged into pyne/pyne.

**Pull request closed without merging**